Moved push notification request (iOS) to Home screen

kwingram25 opened the pull request:
Fixes #3684 

### Summary:
Relocated iOS push notification request to post-login home screen

### Steps to test:
- Run Status on iOS device/simulator
- Create account and login
- Device should display popup asking user to allow push notifications above Home screen

status: ready

kwingram25 committed:
ios, ui, utils: relocated iOS notification permissions to home screen

Signed-off-by: Andrey Shovkoplyas <motor4ik@gmail.com>

flexsurfer commented on src/status_im/ui/screens/home/views.cljs:
```diff
@@ -72,6 +73,11 @@
   (views/letsubs [home-items [:home-items]
                   show-welcome? [:get-in [:accounts/create :show-welcome?]]
                   view-id [:get :view-id]]
+    {:component-did-mount (fn []
```
i'm not sure we can use `did-mount` because this component can be remounted, and also what was the reason to add `platform/ios?` ? i think better to create `fx` and use it in events flow, thanks

kwingram25 commented:
Moved logic into a new re-frame event/handler, dispatched during :account-finalized

I believe the notification prompt only appears on iOS, correct? Removed the platform check since there don't appear to be negative effects from firing on Android, but can put back if needed

flexsurfer commented on src/status_im/ios/core.cljs:
```diff
@@ -31,7 +31,6 @@
          (.hide react/splash-screen))
        :component-did-mount
        (fn []
-         (notifications/request-permissions)
```
why only this one, but not all?

kwingram25 replied:
No reason, it's just setting up logic for when the user does say yes or no at the prompt - it's set up there in the android version so I left it in place and just moved the actual prompt to the home screen

chadyj commented:
Do we know if a user accepts or rejects push notifications?
If so, it would be great to fire a mixpanel event.

flexsurfer commented:
i think yes, @kwingram25 could you please take a look? thanks

kwingram25 commented:
Added granted/denied re-frame events and mixpanel triggers - think it's correct but not sure how to test mixpanel events fire correctly

flexsurfer commented on src/status_im/ui/screens/events.cljs:
```diff
@@ -420,6 +425,11 @@
+(handlers/register-handler-fx
+  :request-notifications
+  (fn [_ _]
+    {::request-notifications-fx {}}))
```
you can pass `nil` here

flexsurfer commented on src/status_im/utils/notifications.cljs:
```diff
@@ -16,9 +16,24 @@
+      (.then
+        (fn [_]
+          (log/debug "notifications-granted")
+          (dispatch [:request-notifications-granted {}]))
```
@jeluard what do you think, is it ok that we need to add events only for mixpanel? maybe we could dispatch something like `[:mixpanel :target-id]` ?

jeluard replied:
I agree it's a little weird but that's probably the best option as it keeps the notification semantic. We may imagine other logic added later (like some popup) or other tracking (like instabug).

flexsurfer commented:
great job @kwingram25 , thanks

flexsurfer commented on src/status_im/utils/mixpanel_events.edn:
```diff
@@ -21,6 +21,14 @@
 
+ ; Push notification settings
+
+ {:label      "Granted push notifications"
```
@chadyj is it ok for you?

chadyj replied:
Wow cool! @lukaszfryc can you test it out?

lukaszfryc replied:
yeah, will do

asemiankevich commented:
screen is moved to the Home tab and both accept and deny events are tracked by mixpanel. No regression found, android is not changed too.

![image](https://user-images.githubusercontent.com/28752642/38201314-8b8056f0-36a0-11e8-89c5-f63841e0412a.png)


![image](https://user-images.githubusercontent.com/28752642/38201408-de7fea0a-36a0-11e8-9d57-b6bbfc367c17.png)

flexsurfer commented:
thanks @kwingram25 could you please squash commits
